### PR TITLE
#159951997 Configure Authors Haven to use PostgreSQL as DB Engine

### PR DESCRIPTION
**Pull request by huxaiphaer**
#### What does this PR do?
* Adds  PostgreSQL Configurations to the application.
#### Description of Task to be completed?
* Ensuring all the environments (Production, Testing, Staging, and Development) are catered for PostgreSQL Engine.

#### How should this be manually tested?
1.  Install PostgreSQL locally on the machine.
2. Set the following variables on your machine. 

-  `DATABASE_NAME`
-  `DATABASE_USER`
-  `DATABASE_PASSWORD`
-  `DATABASE_HOST`
-  `DATABASE_PORT`
-  `DATABASE_TEST_NAME`
-  `DATABASE_URL`
-  `SECRET_KEY`

***Running the application in different environments.***

  - To run in the production environment.
          `python3 manage.py runserver  --settings=authors.settings.development`

  - To run in the Testing environment.
          `python3 manage.py runserver  --settings=authors.settings.testing`

  - To run in Staging environment.
          `python3 manage.py runserver  --settings=authors.settings.staging`

  -  To run in Development environment.
           `python3 manage.py runserver  --settings=authors.settings.development`

#### What are the relevant pivotal tracker stories?
* [#159951997](https://www.pivotaltracker.com/story/show/159951997)

### CODE REVIEW

**Comment by collinewait** [comment]
You can add the .vscode to the gitignore file

**Comment by huxaiphaer** [comment]
Ok @collinewait , let me work on that .

**Comment by reiosantos** [comment]
please squash your commits

**Comment by huxaiphaer** [comment]
@reiosantos , it's done

File: authors/settings/base.py [hunk 1]
```diff
@@ -138,6 +138,18 @@
     'NON_FIELD_ERRORS_KEY': 'error',
 
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        # 'authors.apps.authentication.backends.JWTAuthentication',
+        'authors.apps.authentication.backends.JWTAuthentication',
```
Comment (inno-asiimwe) [inline authors/settings/base.py]: Was this implemented @huxaiphaer , will not break the app? cc @reiosantos

File: authors/settings/local.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from authors.settings.base import *
```
Comment (inno-asiimwe) [inline authors/settings/local.py]: Why do we have an empty file, it does not seem to do anything apart from import everything from base.

File: authors/settings/production.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from authors.settings.base import *
+import dj_database_url
+
+# override base.py here 
+DEBUG = False
+ALLOWED_HOSTS= ['*']
+POSTGRES_URL ="postgres://wmsqvvzzhvqkhg:812a61afbbd7211fb6cbb1cee3f4fd1e4372cb172422467c59e5bc9bb1d01afb@ec2-54-225-76-201.compute-1.amazonaws.com:5432/dc8bll5ofj375l"
```
Comment (inno-asiimwe) [inline authors/settings/production.py]: Put this into an environment variable.

File: authors/settings/staging.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from authors.settings.base import *
```
Comment (inno-asiimwe) [inline authors/settings/staging.py:1]: How different are the staging settings from the production settings?

File: manage.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import sys
 
 if __name__ == "__main__":
-    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "authors.settings")
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "authors.settings.development")
```
Comment (inno-asiimwe) [inline manage.py:6]: Does this change the way django commands are run ? If yes, update the readme to reflect the same.

File: authors/settings/base.py (hunk 1 above)
Comment (huxaiphaer) [reply authors/settings/base.py]: actually the app breaks , i have commented it out @inno-asiimwe , @reiosantos

File: authors/settings/staging.py (hunk 4 above)
Comment (huxaiphaer) [reply authors/settings/staging.py:1]: The heroku postgres connection strings are different for both staging and production, that's the difference.

File: manage.py (hunk 5 above)
Comment (huxaiphaer) [reply manage.py:6]: I have included that in my PR readme

File: authors/settings/staging.py (hunk 4 above)
Comment (inno-asiimwe) [reply authors/settings/staging.py:1]: @huxaiphaer if we are moving the `database_url` to environment variables shall we still need different settings modules?

**Review by inno-asiimwe — APPROVED** [review]

File: manage.py (hunk 5 above)
Comment (huxaiphaer) [reply manage.py]: @inno-asiimwe  no

**Review by inno-asiimwe — APPROVED** [review]

**Pull request merged**